docs/src/pages/docs/functions/logic-functions.md

```diff
@@ -308,6 +467,61 @@ function. Either the original or modified value will be returned depending on
 the result of the predicate. Check out [`unless`](#unless) for a negated version
 of this function.
 
+<!-- eslint-disable no-console -->
+
+```javascript
+import when from 'crocks/logic/when'
+
+import compose from 'crocks/core/compose'
+import composeB from 'crocks/combinator/composeB'
+import constant from 'crocks/combinator/constant'
+import flip from 'crocks/combinator/flip'
+
+// double :: Number -> Number
+const double = a =>
+  a * 2
+
+// doubleWhen :: (a -> Boolean) -> Number -> Number
+const doubleWhen =
+  flip(when, double)
+
+doubleWhen(constant(true), 21)
+//=> 42
+
+doubleWhen(constant(false), 21)
+//=> 21
+
+// gt :: Number -> Number -> Boolean
+const gt = a => b =>
+  b > a
+
+// subtract :: Number -> Number
+const subtract = a => b =>
+  b - a
+
+// protectedSubtract :: Number -> Number -> Number
+const protectedSubtract = composeB(
+  when(gt(0)),
+  subtract
+)
+
+// smallExplosion :: Number -> Number
+const smallExplosion = protectedSubtract(15)
+
+// largeExplosion :: Number -> Number
+const largeExplosion = compose(
+  smallExplosion,
+  smallExplosion,
+  smallExplosion
+)
+
+const startingHealth = 30
```

Comment:
Think that it may be nice to stay in a declairative style, would something like this be an acceptable means of staying :corn:sistant?
```js
import curry from 'crocks/helpers/curry'
import mapProps from 'crocks/helpers/mapProps'

// Player :: { health: Number  }
// player :: Player
const player = {
  health: 30
}

// affectHealth :: (Number -> Number) -> Player -> Player
const affectHealth = curry(
  fn => mapProps({ health: fn })
)

affectHealth(smallExplosion, player)
//=> { health: 15 }

affectHealth(largeExplosion, player)
//=> { health: 0 }
```

Reply:
I like it, I still have a lot to learn in making this FP stuff work nice and look nice :)